core/src/test/scala/sttp/openai/client/SyncClientSpec.scala

```diff
@@ -0,0 +1,81 @@
+package sttp.openai.client
+
+import org.scalatest.EitherValues
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import sttp.client4._
+import sttp.model.StatusCode._
+import sttp.client4.testing._
+import sttp.openai.OpenAIExceptions.OpenAIException
+import sttp.openai.OpenAIExceptions.OpenAIException._
+import sttp.openai.OpenAISyncClient
+import sttp.openai.requests.models.ModelsResponseData._
+
+class SyncClientSpec extends AnyFlatSpec with Matchers with EitherValues {
+  private val testData = List(
+    (TooManyRequests, classOf[RateLimitException]),
+    (BadRequest, classOf[InvalidRequestException]),
+    (NotFound, classOf[InvalidRequestException]),
+    (UnsupportedMediaType, classOf[InvalidRequestException]),
+    (Unauthorized, classOf[AuthenticationException]),
+    (Forbidden, classOf[PermissionException]),
+    (Conflict, classOf[TryAgain]),
+    (ServiceUnavailable, classOf[ServiceUnavailableException]),
+    (Gone, classOf[APIException])
+  )
+  private val errorResponse =
```

Comment:
**suggestion:** Please add a test that verifies if this gets deserialized to proper fields in the exception.